rust/cubesql/src/mysql/mod.rs

```diff
@@ -576,9 +577,41 @@ impl<W: io::Write + Send> AsyncMysqlShim<W> for Backend {
     ) -> Result<(), Self::Error> {
         match self.execute_query(query).await {
             Err(e) => {
-                error!("Error during processing {}: {}", query, e.to_string());
-                results.error(ErrorKind::ER_INTERNAL_ERROR, e.message.as_bytes())?;
+                // @TODO Should this be at the top of this file? Or maybe in some other file?
+                #[derive(Deserialize, Debug)]
+                struct CubeAPIError {
+                    error: String,
+                }
 
+                // Not really sure if this is the best way or place, should revisit this
+                let first_json_brace = e.message.find("{");
```

Comment:
@ovr Thoughts?

Reply:
1. You have chosen the wrong idea & the wrong place for pretty-printing errors from the cube.js. It's an awful idea to search JSON substring from error's message for reformating instead of throwing it correctly.

Native:
https://github.com/cube-js/cube.js/blob/v0.29.7/packages/cubejs-backend-native/src/transport.rs#L119

Standalone:
https://github.com/cube-js/cube.js/blob/v0.29.7/rust/cubesql/src/schema/mod.rs#L66
https://github.com/cube-js/cube.js/blob/v0.29.7/rust/cubeclient/src/apis/default_api.rs#L90